Build issues

make tries to install @vscode/vsce each time it is invoked. That shouldn't be needed.

Also, on Linux it fails if run as ordinary user. That should never happen. People expect to run `make install` using sudo, but not `make` which is supposed to just build the software and not install anything (yet).

If I still try to run `sudo make`, this is what I get:

~~~
Building Pd Remote VS Code...
npm install -g @vscode/vsce

changed 106 packages, and audited 107 packages in 2s

31 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
vsce package
Executing prepublish script 'npm run vscode:prepublish'...

> pd-remote-vscode@1.0.1 vscode:prepublish
> npm run compile

npm WARN logfile Error: EACCES: permission denied, scandir '/root/.npm/_logs'
npm WARN logfile  error cleaning log files [Error: EACCES: permission denied, scandir '/root/.npm/_logs'] {
npm WARN logfile   errno: -13,
npm WARN logfile   code: 'EACCES',
npm WARN logfile   syscall: 'scandir',
npm WARN logfile   path: '/root/.npm/_logs'
npm WARN logfile }

> pd-remote-vscode@1.0.1 compile
> tsc -p ./

sh: line 1: tsc: command not foundile Error: EACCES: permission denied, scandir '/root/.n
 ERROR  npm failed with exit code 127
make: *** [Makefile:7: build] Error 1
~~~

The output at the bottom seems garbled, but it seems to complain both about a missing tsc command, and still missing some permissions in spite of running this as root.

In any case, I'm really worried that it wants root access just to *build* the package. That's a no-go for me.

So the Makefile doesn't seem to work at all on Linux. I'm running this on Arch -- Manjaro, more precisely, with visual-studio-code 1.75.1 installed from the Arch User Repositories, `code` is on PATH. npm is 8.19.2-1 installed from the official Arch repositories.

Trying to just run `vsce package` (without sudo) manually (after `sudo npm install -g @vscode/vsce`) also doesn't work, it complains about that missing tsc command again:

~~~
Executing prepublish script 'npm run vscode:prepublish'...

> pd-remote-vscode@1.0.1 vscode:prepublish
> npm run compile


> pd-remote-vscode@1.0.1 compile
> tsc -p ./

sh: line 1: tsc: command not found
 ERROR  npm failed with exit code 127
~~~

Looks like I probably miss some typescript bits and bobs there? Where can I find that tsc command?

--- a/Makefile
+++ b/Makefile
@@ -1,14 +1,34 @@
 PROJECT = "Pd Remote VS Code"
 VERSION=${shell cat package.json | grep version | cut -d '"' -f 4}
+PACKAGE = pd-remote-vscode-${VERSION}.vsix
 
-build:
-	@echo "Building $(PROJECT)..."
+all: $(PACKAGE)
+
+# This needs to be run once, as root, or nothing will work.
+install-vsce:
+	@echo "Installing vsce..."
 	npm install -g @vscode/vsce
-	vsce package
-	@echo "Done."
 
-install:
+# If it isn't needed any more.
+uninstall-vsce:
+	@echo "Uninstalling vsce..."
+	npm uninstall -g @vscode/vsce
+
+$(PACKAGE): node_modules package.json src/extension.ts
+	@echo "Building $(PROJECT)..."
+	vsce package
+
+node_modules:
+	@echo "Installing dependencies..."
+	npm install
+
+# Do *not* run this as root.
+install: all
 	@echo "Installing $(PROJECT)..."
 	@echo "Version: $(VERSION)"
-	code --install-extension pd-remote-vscode-${VERSION}.vsix
-	@echo "Done."
+	code --install-extension $(PACKAGE)
+
+# Delete build artifacts.
+clean:
+	@echo "Cleaning $(PROJECT)..."
+	rm -rf node_modules out *.vsix

--- a/README.md
+++ b/README.md
@@ -1,14 +1,18 @@
 # pd-remote-vscode README
 
-This extension aims to mirror the features of Albert Graefs Emacs-based [Pd-Remote](https://github.com/agraef/pd-remote) as a Visual Studio Code Extension.
+This extension aims to mirror the features of Albert Gräf's Emacs-based [Pd-Remote](https://github.com/agraef/pd-remote) as a Visual Studio Code Extension.
+
+## Pre-requisites
+
+The @vscode/vsce module needs to be installed globally, or nothing will work. To these ends, run `sudo make install-vsce` OR `sudo npm install -g @vscode/vsce` once.
 
 ## Packaging the extension
 
-### Run `make build` OR
+### Run `make` OR
 
-1. Install VS Codes extension manager vsce from npm  
-```npm install -g @vscode/vsce```
-2. In the repositorys root directory, run  
+1. To install requisite modules using npm, run  
+```npm install```
+2. To package the extension, run  
 ```vsce package```
 
 ## Installing the extension
@@ -30,7 +34,7 @@ This extension aims to mirror the features of Albert Graefs Emacs-based [Pd-Remo
 
 ## Usage
 
-You may use the [examples in Albert Graefs Repository](https://github.com/agraef/pd-remote/tree/main/examples) to test the extension.
+You may use the [examples in Albert Gräf's Repository](https://github.com/agraef/pd-remote/tree/main/examples) to test the extension.
 The default connection is opened on localhost:4711 over UDP. You may change these values in VSCodes workspace or user settings.
 
 ### Supported Commands

